### PR TITLE
Create repos and download/upload test

### PR DESCRIPTION
**Pull request by ruhan1**
This is the download/upload part. @ligangty I add the version change, ie. replace "redhat-###" with "redhat-999999".

### CODE REVIEW

**Comment by sswguo** [comment]
LGTM.

File: pkg/buildtest/run.go [hunk 1]
```diff
@@ -190,61 +191,60 @@ func prepareCacheDirectories() {
...
+// 				get := matches[i][1]
+// 				if strings.HasPrefix(get, "build-") {
+// 					final = strings.ReplaceAll(final, get, buildName)
+// 					break
+// 				}
+// 			}
+// 		}
+// 	}
+// 	return final
+// }
```
Comment (ligangty) [inline pkg/buildtest/run.go:240]: Ok, seems these functions are not needed anymore. Recommend to remove them but not comment them for clear.

File: pkg/buildtest/run_test.go [hunk 2]
```diff
@@ -6,29 +6,29 @@ import (
...
+// 				"http://testdownload/maven/for/test/org/jboss/eap/wildfly-configadmin/7.3.8.GA-redhat-00001/wildfly-configadmin-7.3.8.GA-redhat-00001.pom.md5",
+// 				"http://testdownload/maven/for/test/org/jboss/eap/wildfly-configadmin/7.3.8.GA-redhat-00001/wildfly-configadmin-7.3.8.GA-redhat-00001.pom.sha1"}
+// 			finalDowns := decorateChecksums(testDownloads)
+// 			So(len(finalDowns), ShouldEqual, 6)
+// 		})
 
-	})
-}
+// 	})
+// }
```
Comment (ligangty) [inline pkg/buildtest/run_test.go:27]: Recommend to remove

File: pkg/buildtest/run.go [hunk 3]
```diff
@@ -174,61 +191,60 @@ func prepareCacheDirectories() {
...
-// generate a random 5 digit  number for a build repo like "build-test-xxxxx"
+// generate a random 5 digit  number for a build repo like "build-test-9xxxxx"
 func generateRandomBuildName() string {
-	buildPrefix := "build-test-"
 	rand.Seed(time.Now().UnixNano())
-	min := 10000
-	max := 99999
-	return fmt.Sprintf(buildPrefix+"%v", rand.Intn(max-min)+min)
+	min := 900000
+	max := 999999
```
Comment (ligangty) [inline pkg/buildtest/run.go:246]: Little curious, why 900000~999999 here?

**Review by ligangty — COMMENTED** [review]
LGTM

File: pkg/buildtest/run.go (hunk 3 above)
Comment (ruhan1) [reply pkg/buildtest/run.go:246]: yeah, I want to start from 9##### to easy to catch eyes. I use the save number when replace the versions, e.g, "2.7.0.redhat-9#####".

**Pull request merged**